vllm/executor/gpu_executor.py

```diff
@@ -37,7 +37,10 @@ def __init__(
         self._init_worker()
 
         # Profile the memory usage and initialize the cache.
-        self._init_cache()
+        if not self.model_config.embedding_mode:
+            self._init_cache()
+        else:
+            self._init_embedding()
```

Comment:
can we find a common abstraction for both types?

Reply:
Removed `_init_embedding`